x-pack/plugins/security/public/views/login/components/basic_login_form.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton, EuiCallOut, EuiFieldText, EuiFormRow, EuiPanel, EuiSpacer } from '@elastic/eui';
+import React, { ChangeEvent, Component, Fragment } from 'react';
+import { LoginState } from '../../../../common/login_state';
+
+interface Props {
+  http: any;
+  window: any;
+  infoMessage?: string;
+  loginState: LoginState;
+  next: string;
+}
+
+interface State {
+  hasError: boolean;
+  isLoading: boolean;
+  username: string;
+  password: string;
+  message: string;
+}
+
+export class BasicLoginForm extends Component<Props, State> {
+  public state = {
+    hasError: false,
+    isLoading: false,
+    username: '',
+    password: '',
+    message: '',
+  };
+
+  public componentDidMount() {
+    document.addEventListener('keydown', this.submitOnEnter);
+  }
+
+  public componentWillUnmount() {
+    document.removeEventListener('keydown', this.submitOnEnter);
+  }
+
+  public render() {
+    return (
+      <Fragment>
+        {this.renderMessage()}
+        <EuiPanel>
+          <EuiFormRow label="Username">
+            <EuiFieldText
+              name="username"
+              title="username"
+              data-test-subj="loginUsername"
+              value={this.state.username}
+              onChange={this.onUsernameChange}
+              disabled={this.state.isLoading}
+              aria-required={true}
```

Comment:
Please replace `aria-required={true}` by a `required` attribute. [First rule of ARIA usage](https://www.w3.org/TR/using-aria/#rule1) is: never use ARIA if there is native HTML functionality expressing the same. Usually also the native semantic provides a bit more benefits, e.g. in case of `required` the whole form is "marked" as invalid if a required field is missing, so you could actually use a ref of the form element and check it's validity that way. Also a screenreader is that way able to actually determine the forms state, no matter on which form element it is at the moment.

cc @ryankeairns since I think that suggestion was from you

Reply:
Changing `aria-required` to `required` marks the html form as invalid, which we are styling with a red background image. Since the user hasn't done anything wrong when first visiting the login page, this feels wrong to me. My first instinct is to override the EUI styles for this, but I wanted to get some other opinions before I took that rather blunt approach


![image](https://user-images.githubusercontent.com/3493255/46077364-95da5300-c15e-11e8-9323-9ceb1891db96.png)